pkg/policy: check if EndpointSelector is nil in CNP rule sanitization

ianvernon opened the pull request:
Per the definition of a Rule, EndpointSelector cannot be empty. Ensure this constraint
is met when rules are sanitized.

Signed-off by: Ian Vernon <ian@cilium.io>

**Summary of changes**:

Fixes: #3266 

```release-note
Reject CiliumNetworkPolicy rules which do not have EndpointSelector field
```

Commit:
pkg/policy: check if EndpointSelector is provided in CNP rule

Per the definition of a Rule, EndpointSelector cannot be empty. Ensure this constraint
is met when rules are sanitized.

Signed-off by: Ian Vernon <ian@cilium.io>

tgraf reviewed (approved):
Good catch with the CIDR tests!

Commit:
test/runtime: remove no endpoint selector Ginkgo test

Testing whether policy with no endpoint selector is denied is now done as a
unit test. Thus, having a Ginkgo test for this is not necessary. Delete said
Ginkgo test.

Signed-off by: Ian Vernon <ian@cilium.io>

aanm commented on test/runtime/Policies.go:
```diff
@@ -109,12 +109,6 @@ var _ = Describe("RuntimeValidatedPolicyEnforcement", func() {
-		It("Handles missing required fields", func() {
-			By("Apply a policy with no endpointSelector without crashing")
-			_, err := vm.PolicyImportAndWait(vm.GetFullPath("no_endpointselector_policy.json"), helpers.HelperTimeout)
-			Expect(err).Should(BeNil())
```
We could change this to `Expect(err).Should(Not(BeNil()))`

ianvernon replied:
We already have a unit test for this; there is no reason to have this in the Ginkgo tests.
